### PR TITLE
add normalize_with_delta

### PR DESCRIPTION
**Pull request by tthsqe12**

### CODE REVIEW

**Comment by thofma** [comment]
The keyword argument names (`equidimDec` and `primeDec`) look a bit funny in the Oscar world, but I guess this is Singular legacy?

Also, according to codecov there are some untested branches.

**Comment by fingolfin** [comment]
There is something wrong with coverage generated by `jldoctest`, see issue #323.

File: src/Rings/mpoly-affine-algebras.jl [hunk 1]
```diff
@@ -141,27 +164,30 @@ the algorithm computes an equidimensional decomposition of the radical ideal $I$
...
-      idgens = map(p->_badpolymap(p, A.R), gens(l[2][i]))
-      (newA, hom, (A(idgens[end]), ideal(A, idgens)))
-    end
-    for i in 1:length(l[1])]
+  l = Singular.LibNormal.normal(I.gens.S, _convert_normalize_algorithm(alg))
+  return _convert_normalize_data(A, l)
+end
+
+@doc Markdown.doc"""
+
```
Comment (fingolfin) [inline src/Rings/mpoly-affine-algebras.jl:179]:
```suggestion
    normalize_with_delta(A::MPolyQuo; alg=:equidimDec)

```

**Comment by thofma** [comment]
There is no `jldoctest` added here, just normal tests I think.

File: test/Rings/MPolyQuo.jl [hunk 2]
```diff
@@ -23,6 +23,36 @@ end
...
+    @test parent(FI[1]) == Q
+    @test isa(FI[2], Oscar.Ideal)
+    @test parent(M(Q(x+y))) == S
+    @test isa(delta, Int)
+  end
+
+  for algorithm in (:equidimDec, :primeDec)
+    R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"])
+    Q, _ = quo(R, ideal(R, [(x^2 - y^3)*(x^2 + y^2)*x]))
+    for (S, M, FI) in normalize(Q)
```
Comment (fingolfin) [inline test/Rings/MPolyQuo.jl]: Perhaps this was meant? (I can't test it right now)
```suggestion
    for (S, M, FI) in normalize(Q; alg=algorithm)
```

File: test/Rings/MPolyQuo.jl [hunk 3]
```diff
@@ -23,6 +23,36 @@ end
...
+      @test parent(FI[1]) == Q
+      @test isa(FI[2], Oscar.Ideal)
+      @test parent(M(Q(x+y))) == S
+    end
+
+    stuff, deltas, tot_delta = normalize_with_delta(Q)
```
Comment (fingolfin) [inline test/Rings/MPolyQuo.jl]: Maybe this?
```suggestion
    stuff, deltas, tot_delta = normalize_with_delta(Q; alg=algorithm)
```

**Review by fingolfin — APPROVED** [review]
OK by me, but there are oddities in the test, which @tthsqe12 hopefully can double check?

**Comment by tthsqe12** [comment]
Sorry all, It should be better now.

**Comment by fieker** [comment]
I'll merge it after the tests...

**Pull request merged**